src/lib/list/list.ts

```diff
@@ -175,3 +204,122 @@ export class MdListItem implements AfterContentInit {
+
+  @ContentChildren(MdLine) _lines: QueryList<MdLine>;
+
+  /** Whether the label should appear after or before the checkbox. Defaults to 'after' */
+
+  @Input() checkboxPosition: 'before' | 'after' = 'after';
+
+  @ViewChild('autocheckbox') pCheckbox;
+
+  constructor(private _renderer: Renderer2,
+              private _element: ElementRef,
+              @Optional() private _slist: MdSelectionList,
+              @Optional() navList: MdNavListCssMatStyler,
+              @Optional() selectionListStyler: MdSelectionListCssMatStyler,
+              @Optional() public selectionList: MdSelectionListCheckboxer) {
+    this._isNavList = !!navList;
+    this._isSelectionList = !!selectionListStyler;
+  }
+
+
+  ngAfterContentInit() {
+    this._lineSetter = new MdLineSetter(this._lines, this._renderer, this._element);
+  }
+
+  onchange(): void {
+    console.log('checked or not: ' + this.pCheckbox.state + ', isSelected or not: ' + this.isSelected);
+    if(this.isSelected == false) {
+      this.isSelected = true;
+      this.selectionList.checkedItems.select(this._element.nativeElement);
+    }else {
+      this.isSelected = false;
+
+      this.selectionList.checkedItems.deselect(this._element.nativeElement);
+    }
+    console.log(this.selectionList.checkedItems);
+    console.log('current selectionModule: ' + this.selectionList.checkedItems.selected.length);
+  }
+
+  onKeydown(e: KeyboardEvent): void {
+    console.log('who onkeyDown: ' + this.pCheckbox);
+    if(e.keyCode === 32) {
+      let focusedElement = document.activeElement;
+      console.log(focusedElement === this._element.nativeElement);
+      if(focusedElement === this._element.nativeElement) {
+        if(this.isSelected == false) {
+          this.isSelected = true;
+          this.selectionList.checkedItems.select(this._element.nativeElement);
+        }else {
+          this.isSelected = false;
+
+          this.selectionList.checkedItems.deselect(this._element.nativeElement);
+        }
+        console.log('current selectionModule: ' + this.selectionList.checkedItems.selected.length);
+      }
+    }
+  }
+
+  /** Whether this list item should show a ripple effect when clicked.  */
+  isRippleEnabled() {
+    return !this.disableRipple && (this._isNavList || this._isSelectionList)
+      && !this._slist.disableRipple;
+  }
+
+  _handleFocus() {
+    this._renderer.addClass(this._element.nativeElement, 'mat-list-item-focus');
+  }
+
+  _handleBlur() {
+    this._renderer.removeClass(this._element.nativeElement, 'mat-list-item-focus');
+  }
+
+  /** Retrieves the DOM element of the component host. */
+  _getHostElement(): HTMLElement {
+    return this._element.nativeElement;
+  }
+}
+
+@Directive({
+  selector: 'md-selection-list, mat-selection-list',
+})
+export class MdSelectionListCheckboxer {
+
+  checkedItems: SelectionModel<HTMLElement> = new SelectionModel<HTMLElement>(true);
```

Comment:
`checkedItems` -> `_selectedOptions`

This property should also live directly on `MdSelectionList` (you shouldn't need this class at all)